ggshield/sca/output/text_handler.py

```diff
@@ -1,20 +1,150 @@
+
+
+def populate_incidents_list(
+    result_list: List[SCALocationVulnerability],
+    locations_with_incidents: Dict[str, OutputLocationData],
+    incident_list_key: str,
+    counter_key: str,
+) -> None:
+    for vuln_dep_file in result_list:
+        if vuln_dep_file.location not in locations_with_incidents:
+            locations_with_incidents[vuln_dep_file.location] = OutputLocationData(
+                location=vuln_dep_file.location,
+                added=[],
+                removed=[],
+                added_counter={"critical": 0, "high": 0, "medium": 0, "low": 0},
+                removed_counter={"critical": 0, "high": 0, "medium": 0, "low": 0},
+            )
+
+        # Update counter of vulns to sort locations by number of incidents
+        for pkg_version_with_vuln in vuln_dep_file.package_vulns:
+            for vuln in pkg_version_with_vuln.vulns:
+                incidents_list = locations_with_incidents[
+                    vuln_dep_file.location
+                ].__getattribute__(incident_list_key)
+                incidents_list.append(
+                    OutputIncidentData(
+                        package_full_name=pkg_version_with_vuln.package_full_name,
+                        version=pkg_version_with_vuln.version,
+                        severity=vuln.severity,
+                        summary=vuln.summary,
+                        fixed_version=vuln.fixed_version,
+                        identifier=vuln.identifier,
+                        cve_ids=vuln.cve_ids,
+                    )
+                )
+
+                # Update counter of vulns to sort locations by number of incidents
+                locations_with_incidents[vuln_dep_file.location].__getattribute__(
+                    counter_key
+                )[vuln.severity.lower()] += 1
+
+
+def get_sorted_locations_diff(
+    result: SCAScanDiffOutput, with_removed: bool = True
+) -> Generator[OutputLocationData, None, None]:
+    """
+    Retrieve the locations sorterd by number of incidents for each severity
+    """
+    # First generates a dict of locations with the associated number of incidents
+    locations_with_vulns: Dict[str, OutputLocationData] = {}
+
+    # Handle added vulnerabilities
+    populate_incidents_list(
+        result_list=result.added_vulns,
+        locations_with_incidents=locations_with_vulns,
+        incident_list_key="added",
+        counter_key="added_counter",
+    )
+
+    # Handle removed ones if required
+    if with_removed:
+        populate_incidents_list(
+            result_list=result.removed_vulns,
+            locations_with_incidents=locations_with_vulns,
+            incident_list_key="removed",
+            counter_key="removed_counter",
+        )
+
+    # Finally yield locations in sorted order
+    for vuln_loc in sorted(
+        locations_with_vulns.values(), key=lambda elt: elt.sort_key()
+    ):
+        yield vuln_loc
+
+
+def get_sorted_locations_all(result: SCAScanAllOutput):
```

Comment:
I could somehow, but in the case of `ScanAllOutput`, we have to use `result_list=result.found_package_vulns,` whereas in the case of `ScanDiffOutput`, we need to use `result_list=result.added_vulns` and `result_list=result.removed_vulns`

Reply:
it could be an argument that you pass to the method couldn't it?

Reply:
Indeed, but I find it king of ugly, I already think it's not really good to pass the `counter_key` and `incident_list_key` as I do in `populate_incidents_list`. -> I try an other solution, tell me if you think it's better or not. 
The other point is that it would allow to call `get_sorted_locations_diff` for a `SCAScanAllOutput` with `with_removed=True`, and we don't want this to happen.

Reply:
I pushed my changes, WDYT ?